src/ray/gcs/gcs_client/global_state_accessor.cc

```diff
@@ -240,6 +270,31 @@ bool GlobalStateAccessor::UpdateWorkerDebuggerPort(const WorkerID &worker_id,
   return future.get();
 }
 
+bool GlobalStateAccessor::UpdateWorkerNumPausedTasks(const WorkerID &worker_id,
+                                                     const int num_paused_tasks_delta) {
+  // debugger mutex is used to avoid concurrent updates to the same worker
+  absl::WriterMutexLock debugger_lock(&debugger_tasks_mutex_);
+  std::promise<bool> promise;
+  {
+    absl::ReaderMutexLock lock(&mutex_);
+    RAY_CHECK_OK(gcs_client_->Workers().AsyncUpdateWorkerNumPausedTasks(
+        worker_id, num_paused_tasks_delta, [&promise](const Status &status) {
+          RAY_CHECK_OK(status);
+          promise.set_value(status.ok());
```

Comment:
this will deadlock if this API is called within the same thread as the callback thread (thread_io_service_). 

Can you add a check to the beginning of the function that the current thread is not same as thread_io_service_?

Reply:
done.